src/services/EventService.ts

```diff
@@ -4,37 +4,74 @@ import { getRepository, getConnection } from 'typeorm';
 import { validateOrReject } from 'class-validator';
 import { EventChannel } from '../entities/Channel';
 import { formatValidationErrors, APIError, HttpCode } from '../util/errors';
+import sharp from 'sharp';
+import { writeFile as _writeFile, unlink as _unlink } from 'fs';
+import { promisify } from 'util';
 
-type EventCreationData = Omit<APIEvent, 'id' | 'channelID'>;
+const writeFile = promisify(_writeFile);
+const unlink = promisify(_unlink);
+
+type EventCreationData = Omit<APIEvent, 'id' | 'channelID' | 'image'> & { image: string|boolean };
 
 enum PatchEventError {
 	IdMissing = 'Event ID missing',
-	EventNotFound = 'Event does not exist'
+	EventNotFound = 'Event does not exist',
+	InvalidImage = 'File must be an image.'
 }
 
 @singleton()
 export default class EventService {
-	public async createEvent(data: EventCreationData): Promise<APIEvent> {
+	public async uploadImage(image: string|boolean, event: Event, file?: Express.Multer.File): Promise<Event> {
+		const unsetImage = typeof image === 'boolean' ? !image : image === 'false';
+
+		// If there is an image, try to process it
+		let processedImage: Buffer|undefined;
+		if (!unsetImage && file?.buffer && file.buffer.length > 0) {
+			processedImage = await sharp(file.buffer)
+				.resize({ width: 800, height: 800, fit: sharp.fit.contain })
```

Comment:
```suggestion
				.resize({ width: 800 })
```

Since only admins can use this endpoint, they will be given instructions to use a specific aspect ratio when uploading, so it is fine just to specify a width. If we specified the height too, then I think the image would come out as a square.